components/_Progress-Bar/progress-bar-initialization.js

```diff
@@ -0,0 +1,111 @@
+window.addEventListener('load', function() {
+        
+    //Progress bar Global Definition------------------
+    //Initialize the progress bar with the element id, a progression of 5 steps and set a specific progression step by default      
+    const progressBar = Object.create(window.msdsProgressBar);      
+    const ProgressBarDefinition = document.getElementById("ProgressBarDefinition");
+    if(ProgressBarDefinition) {
+        progressBar.init("ProgressBarDefinition", 5, 4); 
+    }
+
+    //Progress Bar Features scripts------------------    
+    const progressBarFeatures1El = document.getElementById("ProgressBarFeatures1");
+    if(progressBarFeatures1El) {
+        //Initialize the progress bar with 5 steps starting form the default value 0.
+        const progressBarFeatures1 = Object.create(window.msdsProgressBar);
+        progressBarFeatures1.init("ProgressBarFeatures1", 5, 0);
+    }
+        
+    const progressBarFeatures2El = document.getElementById("ProgressBarFeatures2");
+        if(progressBarFeatures2El) {
+        //Initialize the progress bar with 5 steps starting form the 1.
+        const progressBarFeatures2 = Object.create(window.msdsProgressBar);
+        progressBarFeatures2.startFromOne = true; //Set "StartFromOne" variable to true.
+        progressBarFeatures2.init("ProgressBarFeatures2", 5, 0); //Initialize the progress bar with 5 steps.
+    }
+
+    const progressBarFeatures3El = document.getElementById("ProgressBarFeatures3");
+    if(progressBarFeatures3El) {
+        //Set the progress bar to a specific step.
+        const progressBarFeatures3 = Object.create(window.msdsProgressBar);
+        progressBarFeatures3.init("ProgressBarFeatures3", 5, 3); //Initialize the progress bar with 5 steps.
+    }
+
+    //Progress bar Size scripts------------------
+    const progressBarSize1El = document.getElementById('ProgressBarSize1');
+    if(progressBarSize1El) {
+        const progressBarSize1 = Object.create(window.msdsProgressBar);
+        progressBarSize1.init('ProgressBarSize1', 5, 2); //Initialize the progress bar with 5 steps.
+    }    
+            
+    const progressBarSize2El = document.getElementById('ProgressBarSize2');
+    if(progressBarSize2El) {
+        //Set the progress bar to a specific step by default.
+        const progressBarSize2 = Object.create(window.msdsProgressBar);      
+        progressBarSize2.init('ProgressBarSize2', 5, 3); //Initialize the progress bar with 5 steps.
+    }
+     
+    
+    //Light Theme scripts------------------       
+    //Initialize the progress bar with 12 steps.        
+    const lightThemeProgressBarEl = document.getElementById("LightThemeProgressBar");
+    if(lightThemeProgressBarEl) {
+        const lightThemeProgressBar = Object.create(window.msdsProgressBar);
+
+        let lightThemeBaseNumberElement = document.querySelectorAll('#LightThemeBaseNumber')[0]    
```

Comment:
It's not obvious to me why we do this? wouldn't it be easier just to create a variable with 12 in it a then pass that to the init method?

Reply:
I just set this value to the numeric input and use it after to initialize the progress bar. But yes we can do it the way you said as well. same same but different.

Reply:
Cleaning up code is exactly same same but different. First we make it work, then we make it pretty. :)